### PR TITLE
Use nfs-ganesha v5 branch.

### PR DESCRIPTION
**Pull request by james-munson**
https://github.com/longhorn/longhorn/issues/6000

Needs my working branch [longhorn-ganesha-v5-V5.7](https://github.com/james-munson/nfs-ganesha/tree/longhorn-ganesha-v5-V5.7) pushed to longhorn/nfs-ganesha/longhorn-ganesha-v5 first.  (Don't have a PR for that, since the target branch doesn't exist.)

~Then this should be revised to use that tag for the build.~  Done.  See [below](https://github.com/longhorn/longhorn-share-manager/pull/113#issuecomment-1877704257)

### CODE REVIEW

**Comment by innobead** [comment]
@james-munson let's run RWX e2e first to ensure no issues before merging.

File: package/Dockerfile [hunk 1]
```diff
@@ -14,12 +14,12 @@ RUN zypper -n install autoconf bison curl cmake doxygen make git gcc-c++ flex Me
...
 
-RUN curl -L https://github.com/longhorn/nfs-ganesha/archive/refs/tags/v4_20230419.tar.gz| tar zx \
-	  && curl -L https://github.com/nfs-ganesha/ntirpc/archive/v4.2.tar.gz | tar zx \
-	  && mv nfs-ganesha-4_20230419 nfs-ganesha-4.2 \
-	  && rm -r nfs-ganesha-4.2/src/libntirpc \
-	  && mv ntirpc-4.2 nfs-ganesha-4.2/src/libntirpc
-WORKDIR /nfs-ganesha-4.2
+RUN curl -L https://github.com/james-munson/nfs-ganesha/archive/refs/tags/v5.7-trial.tar.gz | tar zx \
+	  && curl -L https://github.com/nfs-ganesha/ntirpc/archive/v5.0.tar.gz | tar zx \
+	  && mv nfs-ganesha-5.7-trial nfs-ganesha-5.7 \
```
Comment (innobead) [inline package/Dockerfile:19]: Why using trial version? We need to use an official release.

**Review by innobead — CHANGES_REQUESTED** [review]

File: package/Dockerfile (hunk 1 above)
Comment (derekbit) [reply package/Dockerfile:19]: Can you submit a PR to https://github.com/longhorn/nfs-ganesha? Thank you.

Comment (innobead) [reply package/Dockerfile:19]: As Derek mentioned, we need to use our downstream version instead like SPDK.

Comment (james-munson) [reply package/Dockerfile:19]: This is a trial build only in my fork, but it is using the V5-stable release.  "Trial" is only for my testing.  It has run e2e in https://ci.longhorn.io/job/private/jog/longhorn-tests-regression/5666 and 5678.

I would love to make a PR, but pushing to what branch?

**Comment by james-munson** [comment]
Updated to pull from longhorn/nfs-ganesha repo's v5 branch at release tag v5_20240104.
Passed a quick run of rwx-related regressions in https://ci.longhorn.io/job/private/job/longhorn-tests-regression/5853/.
Running a full set of regressions now.

**Review by innobead — APPROVED** [review]

**Pull request merged**